packages/alea-frontend/components/EndSemSumAccordion.tsx

```diff
@@ -0,0 +1,229 @@
+      >
+        <Typography variant="h6">End Semester Summary</Typography>
+      </AccordionSummary>
+      <AccordionDetails>
+        <Box>
+           {error && (
+          <Alert severity="error" sx={{ mb: 2 }} onClose={() => setError('')}>
+            {error}
+          </Alert>
+        )}
+
+        {success && (
+          <Alert severity="success" sx={{ mb: 2 }} onClose={() => setSuccess('')}>
+            {success}
+          </Alert>
+        )}
+
+          <Typography variant="body2" color="text.secondary" sx={{ mb: 2 }}>
+            Select quizzes to exclude from the end semester summary calculation:
+          </Typography>
+
+          {loading ? (
+            <Box display="flex" justifyContent="center" alignItems="center" p={2}>
+              <CircularProgress size={24} />
+              <Typography sx={{ ml: 1 }}>Loading quizzes...</Typography>
+            </Box>
+          ) : quizzes.length === 0 ? (
+            <Typography color="text.secondary">No quizzes found for this course.</Typography>
+          ) : (
+            <>
+              <Box sx={{ mb: 2 }}>
+                <Button size="small" variant="outlined" onClick={handleSelectAll} sx={{ mb: 1 }}>
+                  {excludedQuizzes.length === quizzes.length ? 'Deselect All' : 'Select All'}
+                </Button>
+                <Typography variant="body2" color="text.secondary">
+                  {excludedQuizzes.length} of {quizzes.length} quizzes excluded
+                </Typography>
+              </Box>
+
+              <FormGroup sx={{ maxHeight: '300px', overflowY: 'auto', mb: 2 }}>
+                {quizzes.map((quiz) => (
+                  <FormControlLabel
+                    key={quiz.id}
+                    control={
+                      <Checkbox
+                        checked={excludedQuizzes.includes(quiz.id)}
+                        onChange={() => handleQuizToggle(quiz.id)}
+                        size="small"
+                      />
+                    }
+                    label={
+                      <Box>
+                        <Typography variant="body2" component="div">
+                          <SafeHtml html={quiz.title} />
+                        </Typography>
+                        <Typography variant="caption" color="text.secondary">
+                          {quiz.id} • {formatDate(quiz.quizStartTs)}
+                        </Typography>
+                      </Box>
+                    }
+                  />
+                ))}
+              </FormGroup>
+              <Box sx={{ mb: 2 }}>
+                <Typography
+                  variant="body2"
+                  color="primary"
+                  sx={{
+                    mb: 1,
+                    fontWeight: 500,
+                    fontSize: '1rem',
+                    letterSpacing: 0.2,
+                  }}
+                >
+                   Final score is based on top{' '}
+                  <input
```

Comment:
Use MUI's component `TextField` with `type=number`.
In general, we prefer MUI's components for consistent styling and to avoid spending too much time trying to style the component